spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -129,7 +128,88 @@ def write_recording(recording, save_path, acquisition_name):
             description='acquisition_description'
         )
         nwbfile.add_acquisition(ephys_ts)
+
         if os.path.exists(save_path):
             os.remove(save_path)
         with NWBHDF5IO(save_path, 'w') as io:
             io.write(nwbfile)
+
+
+
+
+class NwbSortingExtractor(se.SortingExtractor):
+    def __init__(self, path):
+        try:
+            from pynwb import NWBHDF5IO
+            from pynwb import NWBFile
+            from pynwb.ecephys import ElectricalSeries
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError("To use the Nwb extractors, install pynwb: \n\n"
+                                      "pip install pynwb\n\n")
+        #self._path = path
+        #with NWBHDF5IO(path, 'r') as io:
+        #    nwbfile = io.read()
+        #    NSX = se.NumpySortingExtractor()
+        se.RecordingExtractor.__init__()
+
+    @staticmethod
+    def write_sorting(sorting, save_path):
+        try:
+            from pynwb import NWBHDF5IO
+            from pynwb import NWBFile
+            from pynwb.ecephys import ElectricalSeries
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError("To use the Nwb extractors, install pynwb: \n\n"
+                                      "pip install pynwb\n\n")
+        M = len(sorting.get_unit_ids())
+        fs = sorting.get_sampling_frequency()
+
+        #if NWB files already exists, just adds sorting data to it
+        if os.path.exists(save_path):
+            with NWBHDF5IO(save_path, 'r+') as io:
```

Comment:
Is there a way of setting up the read so that we don't have to repeat code? It might be easier if we stop using the `with` statement and include an `io.close()` statement at that end.